packages/next-server/server/config.ts

```diff
@@ -20,6 +20,7 @@ const defaultConfig: { [key: string]: any } = {
   target: process.env.__NEXT_BUILDER_EXPERIMENTAL_TARGET || 'server',
   poweredByHeader: true,
   compress: true,
+  keepPastBuilds: false,
```

Comment:
I think we should call this something like `legacyKeepPastBuilds` as we'd want to eventually remove the behavior (or support it for a long time but not want people to adopt).